controllers/dscinitialization/servicemesh_setup.go

```diff
@@ -16,7 +16,14 @@ import (
 )
 
 func (r *DSCInitializationReconciler) configureServiceMesh(instance *dsciv1.DSCInitialization) error {
-	switch instance.Spec.ServiceMesh.ManagementState {
+	serviceMeshManagementState := operatorv1.Removed
+	if instance.Spec.ServiceMesh != nil {
```

Comment:
I am not sure if it wouldn't be better to guard this call the same as we do in `removeServiceMesh` below

```golang
if instance.Spec.ServiceMesh == nil {
	return nil
}
```

The current flow when `instance.Spec.ServiceMesh = nil` is following:

- set `serviceMeshManagementState := operatorv1.Removed`
- call `r.removeServiceMesh(instance)`
- in there we check `if instance.Spec.ServiceMesh == nil` and so we do nothing

Therefore I think we can remove the whole defaulting here and it will have no effect anyway. WDYT?

Reply:
Without a pointer we actually face the same issue with "Removed" being now set in DSCI spec. It seems that the intent of "removeServiceMesh" is to only be called from the Finalizer block of DSCI reconcile. This switch case for `operatorv1.Removed` in the original code will also have no effect due to https://github.com/opendatahub-io/opendatahub-operator/blob/ea81e861c457af27c01fa300b3238a44133b0d24/controllers/dscinitialization/servicemesh_setup.go#L55

Reply:
we should work on https://issues.redhat.com/browse/RHOAIENG-6984 even it says to "document" but the main purpose is is to define all different cases and correct the logic in the current code base.  e.g should removeServiceMesh be called in any other scenario if it is not deletion of DSCI